R/estimate_cox.R

```diff
@@ -0,0 +1,181 @@
+    df <- the_lhs()
+    cox_call[["data"]] <- as.symbol(df)
+  } else {
+    df <- as.character(sub("\\[.*$", "", deparse(df_expr))[1])
+  }
+
+  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) {
+    stop(paste0("data can be of class `data.frame` or `tibble` or `data.table`."))
+  }
+
+  data <- as.data.frame(data)
+
+  # Validate columns --------------------------------------------------------
+
+  reqcols <- c(equation, CNSR, AVAL)
+  ss <- c("strata", "tt", "frailty", "ridge", "pspline", "cluster", "I", "factor")
+
+  column_names <- gsub("\\([^()]*\\)", "", reqcols) 
+  column_names <- column_names[!column_names %in% ss]
+  column_names_special <- gsub("\\(([^()]*)\\)|.", "\\1", reqcols, perl=T)
+  column_names_special <- column_names_special[column_names_special!=""]
+  
+  reqcols_fin <- c(column_names, column_names_special)
+ 
+  # Make sure all variables are in dataset
+  if (! all(reqcols_fin %in% colnames(data))){
+    stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols_fin, colnames(data)), collapse = " "), "."))
+  }
+
+  if (!is.numeric(data[[AVAL]])) {
+    stop("Analysis variable (AVAL) is not numeric.")
+  }
+
+  if (!is.numeric(data[[CNSR]])) {
+    stop("Censor variable (CNSR) is not numeric.")
+  }
+
+
+  # Remove NA from the analysis --------------------------------------------
+
+  data <- as.data.frame(data) %>%
+    tidyr::drop_na(AVAL, CNSR)
+
+
+  # Ensure the presence of at least one strata -----------------------------
+
+  if (is.null(equation)) {
+    equation <- "1"
+  } else {
+    equation <- paste(equation, collapse = " + ")
+  }
+
+
+  ## Reverse censoring: see ADaM guidelines versus R survival KM analysis
+  library(survival)
+  formula <- stats::as.formula(glue::glue(paste0("survival::Surv(", AVAL, ", 1-", CNSR, ") ~ {equation}")))
+
+  survfit_object <- survival::coxph(
+    formula, data = data, ...
+  )
+
+
+  # Update Call with original info and dots, similar as update.default ------
+
+  survfit_object$call[[1]] <- quote(survival::coxph)
+  survfit_object$call[["formula"]] <- formula
+  survfit_object$call[["data"]] <- cox_call$data
+  if (length(dots) > 0) {
+    names(survfit_object$call)
+    names(dots)
+    for (i in seq_along(dots)) {
+      survfit_object$call[[names(dots)[i]]] <- unlist(dots[i], use.names = FALSE)
+    }
+  }
+
+
+  # Add additional metadata -------------------------------------------------
+
+  if ("PARAM" %in% colnames(data) && length(setdiff(c("PARAMCD", "PARAM"), equation)) == 2) {
+    # we expect only one unique value => catch mistakes
+    survfit_object[["PARAM"]] <- paste(unique(data[["PARAM"]]), collapse = ", ")
+  }
+
+  if ("PARAMCD" %in% colnames(data) && length(setdiff(c("PARAMCD", "PARAM"), equation)) == 2) {
+    # we expect only one unique value => catch mistakes
+    survfit_object[["PARAMCD"]] <- paste(unique(data[["PARAMCD"]]), collapse = ", ")
+  }
+
```

Comment:
Dont we need to add 'Overall'/Null Model as strata when we dont fill in equation? When we plot the null model this needs to be clear that this is the null model as described in your documentation